src/FakeItEasy/Expressions/ExpressionArgumentConstraintFactory.cs

```diff
@@ -82,7 +82,7 @@ private static void CheckArgumentExpressionIsValid(Expression expression)
         {
             expression = GetExpressionWithoutConversion(expression);
 
-            if (expression is MemberExpression)
+            if (expression is MemberExpression memberExpression && IsMemberOfA(memberExpression.Member))
             {
                 // It's A._, or A.Ignore, or some other property/field, so it's safe.

```

Comment:
I'd consider updating this comment. We originally meant it to refer to `A._` or `A.Ignore` or some field or property on _another type_ (which was an error, as you point out). I'd strike ", or some other property/field".
An alternative would be to say "It's A._, or A.Ignore, or some other property/field on A, so it's safe.".

This isn't a big deal, and if you disagree, I'm content to merge as is.

Reply:
I agree, and I'll change it

Reply:
I amended the commit